nautobot/extras/views.py

```diff
@@ -1487,6 +1491,48 @@ class JobResultView(generic.ObjectView):
     queryset = JobResult.objects.prefetch_related("job_model", "obj_type", "user")
     template_name = "extras/jobresult.html"
 
+    def queryset_to_csv(self, instance):
+        """Format queryset to csv."""
+        csv_data = []
+        headers = JobLogEntry.csv_headers.copy()
+        csv_data.append(",".join(headers))
+
+        for log_entry in instance.logs.all():
+            data = log_entry.to_csv()
+            csv_data.append(csv_format(data))
+
+        return "\n".join(csv_data)
+
+    def get(self, request, *args, **kwargs):
+        """
+        Generic GET handler for accessing an object by PK or slug
+        """
+        instance = get_object_or_404(self.queryset, **kwargs)
+
+        changelog_url = None
+
+        if isinstance(instance, ChangeLoggedModel):
+            changelog_url = instance.get_changelog_url()
+
+        if "export" in request.GET:
+            response = HttpResponse(self.queryset_to_csv(instance), content_type="text/csv")
+            underscore_filename = slugify(instance.job_model).replace("-", "_")
+            filename = f"{underscore_filename}_logs.csv"
```

Comment:
nit: maybe add something like `instance.completed` to the filename as well, to make it easier to distinguish between multiple result exports for the same job?

Reply:
I did some formatting of the `instance.completed`. Let me know your thoughts!